runtime/src/gpu/amd/gpu-amd.c

```diff
@@ -194,7 +204,11 @@ bool chpl_gpu_impl_is_host_ptr(const void* ptr) {
     }
   }
   else {
+#if ROCM_VERSION_MAJOR >= 6
+    return res.type == hipMemoryTypeHost;
```

Comment:
I think this is where the API actually changed with newer ROCms. TLDR: you need `hipMemoryTypeHost || hipMemoryTypeUnregistered`.

In ROCm 5, the "unregistered" memory type didn't even exist: https://rocm.docs.amd.com/projects/HIP/en/docs-5.4.3/doxygen/html/hip__runtime__api_8h.html#aea86e91d3cd65992d787b39b218435a3. I believe the check against `hipSuccess` was what was covering the case where something was just `malloc`ed on the host. With the addition of this "unregistered" type, such allocations now don't really fail when passed to `hipPointerGetAttributes`. IOW, it now returns `hipSuccess`, but sets the memory type to unregistered.

Reverting your other changes to `chpl_gpu_memcpy` to be host-check-based, and `||`ing with that I see that `test/gpu/native/comm/basic.chpl` passes with correct number of comm calls.

I believe with the branch as it stands, your check for device pointers sees some hip-registered host allocations as device allocations. Therefore, more accesses go through the path with device calls, inflating comm counts.